caproto/_backend.py

```diff
@@ -258,15 +256,15 @@ def _preprocess_string_to_wire(values, to_dtype, string_encoding,
         return _preprocess_enum_values(values, to_dtype=ChannelType.INT,
                                        string_encoding=string_encoding,
                                        enum_strings=enum_strings)
-    elif to_dtype in native_int_types:
+    if to_dtype in native_int_types:
         if to_dtype is _LongStringChannelType.LONG_STRING and values:
             if isinstance(values[0], str):
                 return list(encode_or_fail(values[0], string_encoding))
             if isinstance(values[0], bytes):
                 return list(values[0])
-        return [int(v) for v in values]
-    elif to_dtype in native_float_types:
-        return [float(v) for v in values]
+        return [int(v) if v else 0 for v in values]
```

Comment:
I assume this is to account for 0 padded strings?  At what point are we bound to return a certain length?  If we derive tha down-stream from here would it make sense to truncate at the first null byte?

Reply:
It's to account for:
```python
In [1]: int('')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-e4f84bb32d50> in <module>
----> 1 int('')

ValueError: invalid literal for int() with base 10: ''

In [2]: float('')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-45d756431581> in <module>
----> 1 float('')

ValueError: could not convert string to float:
```

EPICS and Python seem to disagree about what empty string to int/float conversion means.

Reply:
interesting,